dvc/dependency/db.py

```diff
@@ -0,0 +1,281 @@
+import logging
+import os
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Any, Dict, Union
+
+from funcy import compact
+
+from dvc.exceptions import DvcException
+from dvc.scm import SCM
+
+from .base import Dependency
+
+if TYPE_CHECKING:
+    from agate import Table
+
+    from dvc.stage import Stage
+
+logger = logging.getLogger(__name__)
+
+
+PARAM_DB = "db"
+PARAM_PROFILE = "profile"
+PARAM_EXPORT_FORMAT = "export_format"  # TODO: kebab-case or snake_case?
+
+
+@contextmanager
+def log_status(msg, log=logger.debug):
+    from funcy import log_durations
+
+    from dvc.ui import ui
+
+    with log_durations(log, msg), ui.status(msg):
+        yield
+
+
+@contextmanager
+def chdir(path):
+    wdir = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(wdir)
+
+
+def export_to(table: "Table", to: str, export_format: str = "csv") -> None:
+    exporter = {"csv": table.to_csv, "json": table.to_json}
+    return exporter[export_format](to)
+
+
+class AbstractDependency(Dependency):
```

Comment:
@iterative/dvc, This is the one that I mentioned in the meeting. 

I don't like this, so consider this as a temporary solution. We can discuss ideas about refactoring here or in the coming meeting and recognize potential solutions to this.

But I am also worried that it'd be a major effort (likely encompassing Stage refactoring and the pipelines).


(also see a lot of skips related to `Stage.is_db_import` below)